front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts

```diff
@@ -126,6 +125,23 @@ async function handler(
       });
     }
     case "DELETE": {
+      const isAuthorized =
+        vault.canWrite(auth) ||
```

Comment:
This one might be false?

Reply:
I don't understand the question, can you clarify? 🙏🏻

Reply:
Sorry. I meant that I'm not sure admins can write here? Checking that it is an admin is enough as this is administration of the vault.

But I guess we decided that can write is true for admins 👍

Reply:
Admin canWrite is false for admins on the system vault, hence this workaround! (Or did I missed something?)

Reply:
Yes right :+1:

I think we should introduce a `canAdmin` to use explicitely for all admin operations. Current code LGTM